edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/EdgeSaslPlainAuthenticator.cs

```diff
@@ -38,9 +38,10 @@ public async Task<IPrincipal> AuthenticateAsync(string identity, string password
                     throw new EdgeHubConnectionException("Identity does not contain device ID.");
                 }
 
-                if (!this.iotHubHostName.Equals(iotHubName))
+                if (!this.iotHubHostName.StartsWith(iotHubName, StringComparison.OrdinalIgnoreCase) ||
+                    this.iotHubHostName[iotHubName.Length] != '.')
                 {
```

Comment:
What is the logic here trying to achieve?

Reply:
Ah yes, should have mentioned - 
the username that is sent from the client is in the form -
user1@sas.testhub1, not user1@sas.testhub1.azure-devices.net
So this checks if the testhub1 is the entire name part of testhub1.azure-devices.net, or more specifically, entire segments in the fully qualified iothub host name (it can be testhub1, or testhub1.azure-devices or something like that, but not testhub or testh or something like that)

Reply:
Perhaps adding a comment on the expectations of identity as you have noted above would be helpful.

Reply:
done